src/main/scala/sangria/marshalling/ArrayMapBuilder.scala

```diff
@@ -62,4 +63,40 @@ class ArrayMapBuilder[T](keys: Seq[String]) {
 
     builder.result()
   }
+
+  lazy val toIterator: Iterator[(String, T)] = {
+    new Iterator[(String, T)] {
+      var index = -1
+      var nextIndex = 0
+
+      @tailrec def nextIndex(current: Int): Option[Int] = {
```

Comment:
What do you think about avoiding `Option` here? `None` result can be represented as `-1`

Reply:
https://github.com/sangria-graphql/sangria-marshalling-api/pull/3/commits/19b1161af3ae9026d925b4fcfdf6e62d36f42ce6